fetch.bs

```diff
@@ -3162,55 +3165,69 @@ the request.
-         <dt>`<code>Save-Data</code>`
-         <dd>a suitable <a href=http://httpwg.org/http-extensions/client-hints.html#save-data>save-data value</a>
-         <dt>`<code>Viewport-Width</code>`
-         <dd>a suitable <a href=http://httpwg.org/http-extensions/client-hints.html#viewport-width>viewport-width value</a>
-         <dt>`<code>Width</code>`
-         <dd>a suitable <a href=http://httpwg.org/http-extensions/client-hints.html#width>width value</a>
-        </dl>
-
-       <li><p>A user agent should <a for="header list">append</a>
-       <var>hintName</var>/<var>value</var> to <var>request</var>'s <a for=request>header list</a>.
-      </ol>
-
-     <li><p>Let <var>record</var> be a new
-     <a for="fetch group">fetch record</a> consisting of
-     <var>request</var> and this instance of the
-     <a for=/>fetch</a> algorithm.
-
-     <li><p>Append <var>record</var> to <var>request</var>'s
-     <a for=request>client</a>'s
-     <a for=fetch>fetch group</a> list of
-     <a for="fetch group">fetch records</a>.
-    </ol>
-  </ol>
+    <li>
+
+     <p><a for=list>For each</a> <var>hintName</var> of <var>request</var>'s
+     <a for=request>client-hints set</a>:
+
+     <ol>
+      <li>
+       <p>Let <var>value</var> be the first matching statement, switching on <var>hintName</var>:
+
+       <dl class=switch>
+        <dt>`<code>DPR</code>`
+        <dd>a suitable <a href=http://httpwg.org/http-extensions/client-hints.html#dpr>dpr value</a>
+        <dt>`<code>Save-Data</code>`
+        <dd>a suitable <a href=http://httpwg.org/http-extensions/client-hints.html#save-data>save-data value</a>
+        <dt>`<code>Viewport-Width</code>`
+        <dd>a suitable <a href=http://httpwg.org/http-extensions/client-hints.html#viewport-width>viewport-width value</a>
+        <dt>`<code>Width</code>`
+        <dd>a suitable <a href=http://httpwg.org/http-extensions/client-hints.html#width>width value</a>
+        <dt>`<code>Device-Memory</code>`
+        <dd>a suitable <a href=https://w3c.github.io/device-memory/#sec-device-memory-client-hint-header>Device-Memory value</a>
+        <dt>`<code>RTT</code>`
+        <dd>a suitable <a href=https://wicg.github.io/netinfo/#rtt-request-header-field>RTT value</a>
+        <dt>`<code>Downlink</code>`
+        <dd>a suitable <a href=https://wicg.github.io/netinfo/#downlink-request-header-field>Downlink value</a>
+        <dt>`<code>ECT</code>`
+        <dd>a suitable <a href=https://wicg.github.io/netinfo/#ect-request-header-field>ECT value</a>
+       </dl>
+
+      <li><p>If <var>request</var> is a <a>subresource request</a> and the result of running <a
+       href="https://wicg.github.io/feature-policy/#should-request-be-allowed-to-use-feature">Should
+       request be allowed to use feature?</a>, given <var>request</var> and <var>hintName</var>’s
+       <a href="http://httpwg.org/http-extensions/client-hints.html#opt-in-via-feature-policy">associated
```

Comment:
Reading the discussion on the relevant PR over in the Client Hints spec (https://github.com/httpwg/http-extensions/pull/696), I don't think the definitions of the policy-controlled features that we're linking to here will be landing there. We need to find a new place for those definitions to live, and change this link.